support-frontend/assets/components/page/offerStrapline.tsx

```diff
@@ -0,0 +1,100 @@
+import { css } from '@emotion/react';
+import { from } from '@guardian/source-foundations';
+import type { ReactElement, ReactNode } from 'react';
+import { offerStraplineBlue } from 'stylesheets/emotion/colours';
+
+const offerStraplineStyles = css`
+	font-family: GuardianTextSans, 'Helvetica Neue', Helvetica, Arial,
+		'Lucida Grande', sans-serif;
+	font-size: 17px;
+	font-weight: bold;
+	padding: 4px 10px 8px;
+	max-width: 100%;
+	width: fit-content;
+	margin-bottom: 0;
+	background-color: ${offerStraplineBlue};
+
+	${from.phablet} {
+		padding: 4px 20px 8px;
+	}
+	${from.tablet} {
+		font-size: 20px;
+		padding: 4px 20px 12px;
+		max-width: 50%;
+	}
+	${from.desktop} {
+		font-size: 24px;
+		max-width: 50%;
+	}
+`;
+
+// Requirement: strapline acts differently (becomes full-width) at smaller device widths if the copy is longer than 32 chars
+const offerStraplineLongCopyStyles = css`
+	font-family: GuardianTextSans, 'Helvetica Neue', Helvetica, Arial,
+		'Lucida Grande', sans-serif;
+	font-size: 17px;
+	font-weight: bold;
+	padding: 4px 10px 8px;
+	width: 100%;
+	margin-bottom: 0;
+	background-color: ${offerStraplineBlue};
+
+	${from.phablet} {
+		max-width: 100%;
+		width: fit-content;
+		padding: 4px 20px 8px;
+	}
+	${from.tablet} {
+		font-size: 20px;
+		padding: 4px 20px 12px;
+		max-width: 50%;
+	}
+	${from.desktop} {
+		font-size: 24px;
+		max-width: 50%;
+	}
+`;
+
+type PropTypes = {
+	copy?: ReactNode;
+};
+
+function OfferStrapline({ copy }: PropTypes): ReactElement {
+	// Requirement: last line must include a minimum of 2 words
+	const noWidowWord = (c: ReactNode) => {
```

Comment:
I don't like this solution for inserting a non-breaking space between penultimate and final words, but don't know how to better attack the problem in a React environment. Suggestions for improvements, or a better approach, very welcome!

Reply:
You need to put the `&nbsp;` within jsx tags, rather than in a string. `dangerouslySetInnerHTML` is not necessary here

Reply:
Thank you - good to get rid of `dangerouslySetInnerHTML`